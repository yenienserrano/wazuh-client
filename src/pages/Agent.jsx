import React from "react";

export const Agent = () => {
  return (
    <div>
      <h1>Hola desde agent</h1>
    </div>
  );
};
