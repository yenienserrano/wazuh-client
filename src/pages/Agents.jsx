import React, { useEffect, useState } from "react";
import { AgentsTable } from "../components/agents/AgentsTable";

export const Agents = () => {
  const [agents, setAgents] = useState([]);

  useEffect(() => {
    fetchData();
  }, []);

  const fetchData = async () => {
    const res = await fetch(`${process.env.REACT_APP_API_URL}/agents`);
    const data = await res.json();
    setAgents(data.data);
  };

  return (
    <div className="p-5">
      <h1 className="text-white">Agents</h1>
      <table className="table table-dark table-striped table-hover">
        <thead>
          <tr>
            <th scope="col">Id</th>
            <th scope="col">name</th>
            <th scope="col">ip</th>
            <th scope="col">Total alerts</th>
          </tr>
        </thead>
        <tbody>
          {agents.map((agent, i) => (
            <AgentsTable key={i} agent={agent} />
          ))}
        </tbody>
      </table>
    </div>
  );
};
