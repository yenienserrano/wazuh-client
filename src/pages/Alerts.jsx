import React, { useEffect, useState } from "react";
import { AlertDetail } from "../components/alerts/AlertDetail";
import { AlertTable } from "../components/alerts/AlertTable";

export const Alerts = () => {
  const [allAlerts, setAllAlerts] = useState(null);
  const [alerts, setAlerts] = useState([]);
  const [offset, setOffset] = useState(0);
  const [limit] = useState(10);
  const [alert, setAlert] = useState(null);
  const [show, setShow] = useState(false);
  const [showBtn, setShowBtn] = useState(true);
  const [filter, setFilter] = useState("withoutFilter");
  const [inputValue, setInputValue] = useState("");

  useEffect(() => {
    fetchAllAlerts();
  }, []);

  useEffect(() => {
    fetchData(offset, limit);
  }, [offset, limit]);

  const fetchData = async (offset, limit) => {
    const res = await fetch(
      `${process.env.REACT_APP_API_URL}/alerts?offset=${offset}&limit=${limit}`
    );
    const data = await res.json();
    setAlerts(data.data);
    setShowBtn(true);
  };

  const fetchAllAlerts = async () => {
    const res = await fetch(`${process.env.REACT_APP_API_URL}/alerts`);
    const data = await res.json();
    setAllAlerts(data.data);
  };

  const prevPage = () => {
    if (offset <= 9) {
      setOffset(0);
    } else {
      setOffset(offset - +limit);
    }
  };

  const nextPage = () => {
    if (+offset >= 90) {
      setOffset(90);
    } else {
      setOffset(+offset + 10);
    }
  };

  const alertDetail = (alert) => {
    setAlert(alert);
    setShow(true);
  };

  const onSubmit = (e) => {
    e.preventDefault();

    if (inputValue.length === 0) {
      handleFilter(filter);
    } else {
      if (alerts.length !== 10) {
        const data = alerts;
        const newData = data.filter(function (item) {
          const itemDataTitle = item._source.agent.name.toUpperCase();
          const itemDataDescp = item._source.syscheck.event.toUpperCase();
          const campo = itemDataTitle + " " + itemDataDescp;
          const textData = inputValue.toUpperCase();
          return campo.indexOf(textData) > -1;
        });
        setAlerts(newData);
        setShowBtn(false);
      } else {
        const data = allAlerts;
        const newData = data.filter(function (item) {
          const itemDataTitle = item._source.agent.name.toUpperCase();
          const itemDataDescp = item._source.syscheck.event.toUpperCase();
          const campo = itemDataTitle + " " + itemDataDescp;
          const textData = inputValue.toUpperCase();
          return campo.indexOf(textData) > -1;
        });
        setAlerts(newData);
        setShowBtn(false);
      }
    }
  };

  const handleFilter = (e) => {
    if (e === "withoutFilter") {
      fetchData(offset, limit);
      setFilter(e);
    } else {
      const data = allAlerts;
      const newData = data.filter(function (item) {
        const itemDataTitle = item._source.agent.name.toUpperCase();
        const itemDataDescp = item._source.syscheck.event.toUpperCase();
        const campo = itemDataTitle + " " + itemDataDescp;
        const textData = e.toUpperCase();
        return campo.indexOf(textData) > -1;
      });
      setAlerts(newData);
      setShowBtn(false);
      setFilter(e);
      setInputValue("");
    }
  };

  return (
    <>
      <div className="p-5">
        <div className="d-flex justify-content-between">
          <h1 className="text-white">Alerts</h1>
          <div className="d-flex">
            <p className="text-white">Filter by agent:</p>
            <select
              className="form-select h-75"
              aria-label="Default select example"
              onChange={(e) => handleFilter(e.target.value)}
            >
              <option value="withoutFilter">Without filter</option>
              <option value="Ubuntu">Ubuntu</option>
              <option value="Debian">Debian</option>
              <option value="ip-10-0-0-180.us-west-1.compute.internal">
                ip-10-0-0-180.us-west-1.compute.internal
              </option>
              <option value="Amazon">Amazon</option>
              <option value="RHEL7">RHEL7</option>
              <option value="Centos">Centos</option>
              <option value="Windows">Windows</option>
            </select>
          </div>
          <form className="input-group mb-3 w-25">
            <input
              type="text"
              className="form-control"
              placeholder="Search"
              value={inputValue}
              aria-describedby="button-addon2"
              onChange={(e) => setInputValue(e.target.value)}
            />
            <button
              className="btn btn-outline-light"
              type="submit"
              id="button-addon2"
              onClick={onSubmit}
              onSubmit={(e) => onSubmit(e)}
            >
              Search
            </button>
          </form>
        </div>
        <table className="table table-dark table-striped table-hover">
          <thead>
            <tr>
              <th scope="col">Id</th>
              <th scope="col">Agent</th>
              <th scope="col">Rule</th>
              <th scope="col">Event</th>
            </tr>
          </thead>
          <tbody>
            {alerts.map((alert) => (
              <AlertTable
                key={alert._id}
                alert={alert}
                alertDetail={alertDetail}
              />
            ))}
          </tbody>
        </table>
        {showBtn && (
          <div className="w-100 d-flex justify-content-between">
            <div className="d-flex justify-content-start">
              {+offset <= 0 ? null : (
                <button className="btn btn-outline-light" onClick={prevPage}>
                  Prev
                </button>
              )}
            </div>
            <div className="d-flex justify-content-end">
              {+offset >= 90 ? null : (
                <button className="btn btn-outline-light" onClick={nextPage}>
                  Next
                </button>
              )}
            </div>
          </div>
        )}
      </div>
      {show && <AlertDetail setShow={setShow} alert={alert} />}
    </>
  );
};
