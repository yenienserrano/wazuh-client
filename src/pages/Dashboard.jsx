import React, { useEffect, useState } from "react";
import { GraficD3 } from "../components/dashboard/GraficD3";

export const Dashboard = () => {
  const [agents, setAgents] = useState(null);
  const [rules, setRules] = useState(null);

  useEffect(() => {
    getAllAgents();
    getAllRules();
    // eslint-disable-next-line
  }, []);
  const getAllAgents = async () => {
    const res = await fetch(`${process.env.REACT_APP_API_URL}/agents`);
    const data = await res.json();
    setAgents(data.data);
  };

  const getAllRules = async () => {
    const res = await fetch(`${process.env.REACT_APP_API_URL}/rules`);
    const data = await res.json();
    setRules(data.data);
  };

  return (
    <div className="text-center">
      <h1 className="text-white mt-3 mb-3">Dashboard</h1>
      <div className="container">
        <div className="row">
          {agents && (
            <div className="col-md-6 d-flex flex-column align-items-center animate__animated animate__fadeIn">
              <h1 className="text-white">Agents alerts</h1>
              <GraficD3 data={agents} />
              <p className="text-white mt-3">
                The number of alerts is reflected on the Y axis and the agent
                id's on the X axis
              </p>
            </div>
          )}
          {rules && (
            <div className="col-md-6 d-flex flex-column align-items-center animate__animated animate__fadeIn">
              <h1 className="text-white">Rules alerts</h1>
              <GraficD3 data={rules} />
              <p className="text-white mt-3">
                The number of alerts is reflected on the Y axis and the rules
                id's on the X axis
              </p>
            </div>
          )}
        </div>
      </div>
    </div>
  );
};
