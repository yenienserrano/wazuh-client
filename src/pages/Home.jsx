import React from "react";

export const Home = () => {
  return (
    <div className="d-flex h-75 justify-content-center align-items-center">
      <h1 className="text-white">Wazuh Client Home</h1>
    </div>
  );
};
