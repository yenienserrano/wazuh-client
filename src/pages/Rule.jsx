import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { AlertDetail } from "../components/alerts/AlertDetail";
import { AlertTable } from "../components/alerts/AlertTable";

export const Rule = () => {
  const [rule, setRule] = useState([]);
  const [alert, setAlert] = useState(null);
  const [alerts, setAlerts] = useState(null);
  const [tenAlerts, setTenAlerts] = useState(null);
  const [show, setShow] = useState(false);
  const [showMore, setShowMore] = useState(false);
  const { id } = useParams();

  useEffect(() => {
    fetchData();
    // eslint-disable-next-line
  }, []);
  const fetchData = async () => {
    const res = await fetch(`${process.env.REACT_APP_API_URL}/rules/${id}`);
    const data = await res.json();
    setRule(data.data);
    setAlerts(data.data.alerts);
    setTenAlerts(data.data.alerts.slice(0, 10));
  };
  const alertDetail = (alert) => {
    setAlert(alert);
    setShow(true);
  };

  return (
    <>
      <div className="container">
        <div className="row">
          <div className="col-md-4">
            <h1 className="text-white mt-4 mb-4">Rule details</h1>
            <div className="card bg-dark text-white">
              <div className="card-header">
                <b>Id:</b> {rule.id}
              </div>
              <ul className="list-group list-group-flush">
                <li className="list-group-item bg-dark text-white">
                  <b>Description:</b> {rule.description}
                </li>
                <li className="list-group-item bg-dark text-white">
                  <b> Level:</b> {rule.level}
                </li>
                <li className="list-group-item bg-dark text-white">
                  <b> Total alerts:</b> {rule.total_alerts}
                </li>
                <li className="list-group-item bg-dark text-white">
                  <b>Gpg13:</b> {rule.gpg13}
                </li>
                <li className="list-group-item bg-dark text-white">
                  <b>Gdpr:</b> {rule.gdpr}
                </li>
                <li className="list-group-item bg-dark text-white">
                  <b>Firedtimes:</b> {rule.firedtimes}
                </li>
              </ul>
            </div>
          </div>
          <div className="col-md-8">
            <h1 className="text-white mt-4 mb-4">Agent alerts</h1>

            <table className="table table-dark table-striped table-hover">
              <thead>
                <tr>
                  <th scope="col">Id</th>
                  <th scope="col">Agent</th>
                  <th scope="col">Rule</th>
                  <th scope="col">Event</th>
                </tr>
              </thead>
              <tbody>
                {alerts &&
                  showMore &&
                  alerts.map((alert) => (
                    <AlertTable
                      key={alert._id}
                      alert={alert}
                      alertDetail={alertDetail}
                    />
                  ))}
                {tenAlerts &&
                  !showMore &&
                  tenAlerts.map((alert) => (
                    <AlertTable
                      key={alert._id}
                      alert={alert}
                      alertDetail={alertDetail}
                    />
                  ))}
              </tbody>
            </table>
            <div className="d-flex justify-content-center mb-5">
              {alerts && alerts.length > 10 ? (
                showMore ? (
                  <button
                    className="btn btn-dark"
                    onClick={() => setShowMore(false)}
                  >
                    See less
                  </button>
                ) : (
                  <button
                    className="btn btn-dark"
                    onClick={() => setShowMore(true)}
                  >
                    See all
                  </button>
                )
              ) : null}
            </div>
          </div>
        </div>
      </div>

      {show && <AlertDetail setShow={setShow} alert={alert} />}
    </>
  );
};
