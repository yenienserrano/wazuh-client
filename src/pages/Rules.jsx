import React, { useEffect, useState } from "react";
import { RulesTable } from "../components/rules/RulesTable";

export const Rules = () => {
  const [rules, setRules] = useState([]);
  const [offset] = useState(0);
  const [limit] = useState(10);

  useEffect(() => {
    fetchData();
  }, [offset, limit]);

  const fetchData = async () => {
    const res = await fetch(`${process.env.REACT_APP_API_URL}/rules`);
    const data = await res.json();
    setRules(data.data);
  };

  return (
    <div className="p-5">
      <h1 className="text-white">Rules</h1>
      <table className="table table-dark table-striped table-hover">
        <thead>
          <tr>
            <th scope="col">Id</th>
            <th scope="col">Level</th>
            <th scope="col">Description</th>
            <th scope="col">Total alerts</th>
          </tr>
        </thead>
        <tbody>
          {rules.map((rule) => (
            <RulesTable key={rule.id} rule={rule} />
          ))}
        </tbody>
      </table>
    </div>
  );
};
