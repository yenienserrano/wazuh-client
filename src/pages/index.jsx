import { Agent } from "./Agent";
import { Agents } from "./Agents";
import { Alerts } from "./Alerts";
import { Home } from "./Home";
import { Rule } from "./Rule";
import { Rules } from "./Rules";
import { Dashboard } from "./Dashboard";

export { Agent, Agents, Home, Rule, Rules, Alerts, Dashboard };
