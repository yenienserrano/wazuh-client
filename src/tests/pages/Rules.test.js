import React from "react";
import { shallow } from "enzyme";
import { Rules } from "../../pages/Rules";

describe("Test in <Rules />", () => {
  const wrapper = shallow(<Rules />);
  const h1 = wrapper.find("h1");
  test("should display the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should show an h1 with the text "Rules"', () => {
    expect(h1.text().trim()).toBe("Rules");
  });
  test("should show an h1 with class text-white", () => {
    expect(h1.prop("className")).toContain("text-white");
  });
});
