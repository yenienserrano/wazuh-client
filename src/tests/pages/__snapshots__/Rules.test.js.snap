// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Rules /> should display the component correctly 1`] = `
<div
  className="p-5"
>
  <h1
    className="text-white"
  >
    Rules
  </h1>
  <table
    className="table table-light table-striped table-hover"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Id
        </th>
        <th
          scope="col"
        >
          Level
        </th>
        <th
          scope="col"
        >
          Description
        </th>
        <th
          scope="col"
        >
          Total alerts
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
